File: maratona-sbc-2016/2238/solve.py
import math
A, B, C, D = input().split()
A = int(A)
B = int(B)
C = int(C)
D = int(D)
ans = -1
x = A
sqrt_c = int(math.sqrt(C))
i = 0
while i <= sqrt_c:
    if (x % B) and not(C % x) and (D % x) :
        ans = x
        break
    x += A
    i += 1

print(ans)

File: maratona-sbc-2016/2238/solve.go
package main
import (
	"fmt"
	"math"
)

func main() {
    var A, B, C, D, x, ans, i, sqrt_c int64
    ans = -1
    fmt.Scanln(&A, &B, &C, &D)
	i = 0
	sqrt_c = int64(math.Sqrt(float64(C)))
	for i <= sqrt_c {
		if (x % B) != 0 && (C % x) == 0 && (D % x) != 0 {
			ans = x
			break
		}
		i += 1
		x += A
	}
	fmt.Println(ans)
}